src/aap_eda/core/management/commands/scheduler.py

```diff
@@ -149,4 +152,47 @@ def handle(self, *args, **options) -> None:
         add_startup_jobs(scheduler)
         add_periodic_jobs(scheduler)
         add_cron_jobs(scheduler)
-        super().handle(*args, **options)
+        # We are going to start our own loop here to catch exceptions which
+        # might be coming from a redis cluster and retrying things.
+        while True:
+            try:
+                super().handle(*args, **options)
+            except (
+                redis.exceptions.TimeoutError,
+                redis.exceptions.ClusterDownError,
+                redis.exceptions.ConnectionError,
+            ) as e:
+                # If we got one of these exceptions but are not on a Cluster go
+                # ahead and raise it normally.
+                if type(self.connection) is not DABRedisCluster:
```

Comment:
Using 'not isinstance(self.connection, DABRedisCluster)' would provide some future-proofing.